core/src/main/java/com/scalar/db/transaction/jdbc/JdbcTransaction.java

```diff
@@ -81,14 +89,17 @@ public List<Result> scan(Scan scan) throws CrudException {
   public void put(Put put) throws CrudException {
     put = copyAndSetTargetToIfNot(put);
 
-    // Ignore the condition in the put
-    if (put.getCondition().isPresent()) {
-      logger.warn("ignoring the condition of the mutation: {}", put);
-      put.withCondition(null);
-    }
-
     try {
-      jdbcService.put(put, connection);
+      if (!jdbcService.put(put, connection)) {
+        // To match the Consensus Commit transaction behavior, instead of throwing an exception here
```

Comment:
> To match the Consensus Commit transaction behavior

Could you elaborate this? It would be great if you tell me an example case that violates the Consensus Commit transaction behavior.

Reply:
For Consensus Commit transaction, the mutational operation condition are evaluated in `tx.commit()`
Though, for JdbcTransaction, the mutational operation conditions are evaluated directly in the `tx.put()` or `tx.delete()`, so before the transaction is committed. 

In the code above, when `jdbcService.put(put, connection)` result is false, it means the `put` was not applied because its condition is not validated. So technically, we could directly throw an exception when `tx.put()` is executed. But we choose not to do that because it means the user would have to handle the "mutation condition unsatisfied exception" differently when using Jdbc Transaction or Consensus Commit Transaction.

Reply:
Thanks for the explanation! Understood.